src/main/webapp/pages/bookmarkedPlaceGuides_initPage.js

```diff
@@ -14,13 +14,15 @@ function initPage() {
     if (!userAuthenticationStatus.isLoggedIn) {
       location.replace(userAuthenticationStatus.loginUrl);
     } else {
-      const menu = new Menu(Menu.PAGE_NAMES.BOOKMARKED_PLACEGUIDES);
-      const mapWidget = new MapWidget();
-      mapWidget.addGeolocationFunctionality();
-      mapWidget.addSearchingFunctionality();
-      map = mapWidget.map;
-      placeGuideManager = new PlaceGuideManager(
-          PlaceGuideManager.PAGE.BOOKMARKED_PLACEGUIDES, map);
+      saveUserInDatabase().then((response) => {
```

Comment:
Isn't this always saving it even if it is already saved?

Reply:
No. The check for the user's existence in the database was previously moved to the server-side.

Reply:
But it still does a network request every time before proceeding, right? Could this delay be avoided or is it necessary?

Reply:
It is necessary due to the way the servlets are organised: it has been always assumed that the current user exists in the database. 
This could be avoided in two ways: adding additional checks and handling the case of the user not being present in the database separately, or by creating a single-page application(which would reduce the number of such network requests to 1/session). 
However, given the time constraint that I have for the project, this is the only feasible solution at the moment.